lib/mongodb.js

```diff
@@ -768,45 +768,100 @@ function convertToMeters(distance, unit) {
   }
 }
 
-function buildNearFilter(query, near) {
-  var coordinates = {};
-  if (typeof near.near === 'string') {
-    var s = near.near.split(',');
-    coordinates.lng = Number(s[0]);
-    coordinates.lat = Number(s[1]);
-  } else {
-    coordinates = near.near;
+function buildNearFilter(query, params) {
+  if (!Array.isArray(params)) {
+    params = [params];
   }
 
-  query.where[near.key] = {
-    near: {
-      $geometry: {
-        coordinates: [coordinates.lng, coordinates.lat],
-        type: 'Point',
+  params.forEach(function(near) {
+    var coordinates = {};
+
+    if (typeof near.near === 'string') {
+      var s = near.near.split(',');
+      coordinates.lng = Number(s[0]);
+      coordinates.lat = Number(s[1]);
+    } else {
+      coordinates = near.near;
+    }
+
+    var props = ['maxDistance', 'minDistance'];
+    //use mongodb default unit 'meters' rather than 'miles'
+    var unit = near.unit || 'meters';
+
+    var queryValue = {
+      near: {
+        $geometry: {
+          coordinates: [coordinates.lng, coordinates.lat],
+          type: 'Point',
+        },
       },
-    },
-  };
+    };
+
+    props.forEach(function(p) {
+      if (near[p]) {
+        queryValue['near']['$' + p] = convertToMeters(near[p], unit);
+      }
+    });
+
+    var property;
 
-  var props = ['maxDistance', 'minDistance'];
-  //use mongodb default unit 'meters' rather than 'miles'
-  var unit = near.unit || 'meters';
-  props.forEach(function(p) {
-    if (near[p]) {
-      query.where[near.key]['near']['$' + p] = convertToMeters(near[p], unit);
+    if (near.key) {
+      if (Array.isArray(near.key)) {
+        property = query.where;
+
+        for (var i = 0; i < near.key.length; i++) {
+          var subKey = near.key[i];
+          if (near.key.hasOwnProperty(i + 1)) {
+            if (!property.hasOwnProperty(subKey)) {
+              property[subKey] = Number.isInteger(near.key[i + 1]) ? [] : {};
+            }
+
+            property = property[subKey];
+          }
+        }
+
+        property[near.key[i - 1]] = queryValue;
+      } else {
+        property = query.where[near.key] = queryValue;
```

Comment:
The `property = ` part isn't very useful, it's just to stay coherent with the code above.